tests/Kiota.Builder.Tests/Writers/Java/CodeMethodWriterTests.cs

```diff
@@ -1502,7 +1502,6 @@ public void WritesSerializerBody()
         Assert.Contains("writeCollectionOfPrimitiveValues", result);
         Assert.Contains("writeCollectionOfObjectValues", result);
         Assert.Contains("writeEnumValue", result);
-        Assert.Contains("writeAdditionalData(this.getAdditionalData());", result);
```

Comment:
can you provide more context on why this is being removed? it feels like we should keep it

Reply:
Via the change we add the additionalData serialization through the accessedProperty not on the class properties themselves, the test covers the case where the property is added, however that will not lead to the serialization since the getter is not present. The correct test would be to add the additionalData getter and show that the serializer will in fact generate this. I think the serializer method should rely on the accessedProperties of the available getters as this will ensure that the getSomeProperty() method will be available. For context this is the change:
```Java
        var additionalDataProperty = parentClass.GetPropertyOfKind(CodePropertyKind.AdditionalData);

        if (additionalDataProperty != null)
            writer.WriteLine($"writer.writeAdditionalData(this.get{additionalDataProperty.Name.ToFirstCharacterUpperCase()}());");
```
I considered the following where we actually have both conditions, let me know your thoughts
```Java
if (parentClass.Methods.Any(x => x.AccessedProperty?.IsOfKind(CodePropertyKind.AdditionalData) ?? false) || 
(parentClass.Properties.Any(x => x.IsOfKind(CodePropertyKind.AdditionalData) ?? false))
```